include/ck/tensor_operation/gpu/grid/gridwise_elementwise_layernorm_welford_variance.hpp

```diff
@@ -0,0 +1,499 @@
+            p_beta_global, beta_grid_desc_m_k.GetElementSpaceSize());
+
+        auto threadwise_welford       = ThreadwiseWelford();
+        threadwise_welford.max_count_ = GetKPerThread(c_grid_desc_m_k, thread_k_cluster_id);
+
+        static_for<0, MThreadSliceSize, 1>{}([&](auto I) {
+            mean_thread_buf(I) = type_convert<AccDataType>(0.0f);
+            var_thread_buf(I)  = type_convert<AccDataType>(0.0f);
+        });
+
+        for(index_t reducedTiles = 0; reducedTiles < num_k_block_tile_iteration; ++reducedTiles)
+        {
+            static_for<0, CThreadBufferNumber, 1>{}([&](auto iK0) { // Thread buffer number
+                static_for<0, NumInput, 1>{}([&](auto I) {          // input load loop
+                    in_global_load_tuple(I).Run(in_grid_2d_desc_tuple[I],
+                                                in_global_buf_tuple[I],
+                                                thread_buffer_desc_m_k,
+                                                make_tuple(I0, I0),
+                                                in_thread_buf_tuple(iK0)(I));
+
+                    in_global_load_tuple(I).MoveSrcSliceWindow(in_grid_2d_desc_tuple[I],
+                                                               thread_copy_fwd_step_m_k);
+                });
+
+                static_for<0, MThreadSliceSize, 1>{}([&](auto iM) { // input add loop
+                    static_for<0, XSrcVectorSize, 1>{}([&](auto iK1) {
+                        constexpr auto offset_m_k =
+                            thread_buffer_desc_m_k.CalculateOffset(make_tuple(iM, iK1));
+
+                        // get reference to in data
+                        const auto in_data_refs = generate_tie(
+                            // return type should be lvalue
+                            [&](auto I) -> const auto& {
+                                return in_thread_buf_tuple(iK0)(I)(Number<offset_m_k>{});
+                            },
+                            Number<NumInput>{});
+
+                        // get reference to dst data
+                        auto out_data_refs = generate_tie(
+                            // return type should be lvalue
+                            [&](auto) -> auto& { return c_thread_buf(iK0)(Number<offset_m_k>{}); },
+                            I1);
+
+                        unpack2(elementwise_op, out_data_refs, in_data_refs);
+                    });
+                });
+                threadwise_welford.Run(c_thread_buf[iK0], mean_thread_buf, var_thread_buf);
+
+                if constexpr(!SweepOnce)
+                {
+                    threadwise_c_store.Run(thread_buffer_desc_m_k,
+                                           make_tuple(I0, I0),
+                                           c_thread_buf(iK0),
+                                           c_grid_desc_m_k,
+                                           c_global_val_buf);

```

Comment:
Why don't you write the result of elementwise into LDS?
It is same as two kernels method for non Sweeponce case

Reply:
Yes, you are right, I'll change it, make the results into LDS.
Thanks for advice!